swarm64_tpc_toolkit/correctness.py

```diff
@@ -21,29 +22,30 @@ def get_correctness_filepath(self, query_id):
         filepath = os.path.join(self.correctness_results_folder, f'{query_id}.csv')
         return filepath
 
-    def has_differences(self, truth, result):
-        self.truth = truth
-        self.result = result
-
-        for truth_idx, truth_row in truth.iterrows():
-            for result_idx, result_row in self.result.iterrows():
-                row_equal = True
-                for column in truth:
-                    if truth[column].dtype == 'float64':
-                        if not numpy.isclose(truth_row[column], result_row[column], rtol=1e-12, atol=0.01):
-                            row_equal = False
-                    else:
-                        if truth_row[column] != result_row[column]:
-                            row_equal = False
-                if row_equal:
-                    self.truth.drop(index=truth_idx, inplace=True)
-                    self.result.drop(index=result_idx, inplace=True)
+    @classmethod
+    def check_for_mismatches(cls, truth, result):
+        differences = result[~result.isin(truth).all(1)]
```

Comment:
why  the `all(1)`?

Reply:
It determines the direction of checking. Here is column wise. Since rows are sorted this should be fine. Will add more tests.